sopel/bot.py

```diff
@@ -554,62 +580,101 @@ def dispatch(self, pretrigger):
 
-                    # check intents
-                    if hasattr(func, 'intents'):
-                        if not intent:
-                            continue
+        Parsed message (or ``pretrigger``) are used to get triggered callables:
+        it will retrieve them by order of priority, and run them. It runs
+        triggered callables in separate threads, unless they are marked
+        otherwise with the :func:`sopel.module.thread` decorator.
 
-                        match = any(
-                            func_intent.match(intent)
-                            for func_intent in func.intents
-                        )
-                        if not match:
-                            continue
+        However, it won't run triggered callables at all when they can't be run
+        for blocked nickname or hostname.
 
-                    # check echo-message feature
-                    if is_echo_message and not func.echo:
-                        continue
+        .. seealso::
 
-                    # check blocked nick/host
-                    # done after we know the trigger would have matched so we
-                    # don't spam logs with "prevented from using" entries about
-                    # functions that weren't going to run anyway
-                    if blocked and not func.unblockable and not trigger.admin:
-                        function_name = "%s.%s" % (
-                            func.__module__, func.__name__
-                        )
-                        list_of_blocked_functions.append(function_name)
-                        continue
+            To get a list of triggered callables by priority, it uses
+            :meth:`~get_triggered_callables`. This method is also responsible
+            for telling ``dispatch`` if the function is blocked or not.
+        """
+        # nickname/hostname blocking
+        nick_blocked = host_blocked = self._is_pretrigger_blocked(pretrigger)
+        blocked = bool(nick_blocked or host_blocked)
+        list_of_blocked_functions = []
 
-                    # call triggered function
-                    wrapper = SopelWrapper(self, trigger)
-                    if func.thread:
-                        targs = (func, wrapper, trigger)
-                        t = threading.Thread(target=self.call, args=targs)
-                        t.start()
-                    else:
-                        self.call(func, wrapper, trigger)
+        # trigger by priority
+        for priority in ('high', 'medium', 'low'):
+            items = self.get_triggered_callables(priority, pretrigger, blocked)
+            for func, trigger, is_blocked in items:
+                # check blocked nick/host
+                # done after we know the trigger would have matched so we
+                # don't spam logs with "prevented from using" entries about
+                # functions that weren't going to run anyway
```

Comment:
This comment block is kind of redundant now. It's probably sufficient to just say:

```py
                # skip running blocked functions, but track them for logging
```

Reply:
I thought you would mention it, but I wasn't sure so I didn't deleted it directly. I'll do that now.